sound/soc/sof/ipc4-pcm.c

```diff
@@ -314,9 +315,99 @@ static int sof_ipc4_pcm_hw_params(struct snd_soc_component *component,
 	return sof_ipc4_build_time_info(sdev, &spcm->stream[substream->stream]);
 }
 
+static int sof_ipc4_get_stream_start_offset(struct snd_sof_dev *sdev,
+					    struct snd_pcm_substream *substream,
+					    struct snd_sof_pcm_stream *stream,
+					    struct sof_ipc4_timestamp_info *time_info)
+{
+	struct sof_ipc4_copier *host_copier = time_info->host_copier;
+	struct sof_ipc4_copier *dai_copier = time_info->dai_copier;
+	struct sof_ipc4_pipeline_registers ppl_reg;
+	u64 stream_start_position;
+	u32 dai_sample_size;
+	u32 ch, node_id;
+	u32 offset;
+
+	node_id = SOF_IPC4_NODE_INDEX(host_copier->data.gtw_cfg.node_id);
+	offset = offsetof(struct sof_ipc4_fw_registers, pipeline_regs) + node_id * sizeof(ppl_reg);
+	sof_mailbox_read(sdev, sdev->fw_info_box.offset + offset, &ppl_reg, sizeof(ppl_reg));
+	if (ppl_reg.stream_start_offset == SOF_IPC4_INVALID_STREAM_POSITION)
+		return -EBUSY;
```

Comment:
why not -EINVAL?

There's no real reason to choose any specific code anyways, since you squelch the error below.

Reply:
This means FW is still preparing for stream_start_offset and it is not ready